src/instanceMethods/get-template-params.js

```diff
@@ -0,0 +1,109 @@
+import defaultParams from '../utils/params.js'
+import { toArray, capitalizeFirstLetter } from '../utils/utils.js'
+
+export const getTemplateParams = (params) => {
+  const template = typeof params.template === 'string' ? document.querySelector(params.template) : params.template
+  if (!template) {
+    return {}
+  }
+  const templateContent = template.content || template // IE11
+
+  const result = Object.assign(
+    getSwalParams(templateContent),
+    getSwalButtons(templateContent),
+    getSwalImage(templateContent),
+    getSwalIcon(templateContent),
+    getSwalInput(templateContent),
+    getSwalStringParams(templateContent, ['title', 'html', 'footer']),
+  )
+  return result
+}
+
+const getSwalParams = (templateContent) => {
+  const result = {}
+  toArray(templateContent.querySelectorAll('swal-param')).forEach((param) => {
+    const paramName = param.getAttribute('name')
+    let value = param.getAttribute('value')
+    if (typeof defaultParams[paramName] === 'boolean' && value === 'false') {
+      value = false
+    }
```

Comment:
```suggestion
    if (typeof defaultParams[paramName] === 'boolean') {
      value = JSON.parse(value)
    }
```

IMO values like "fals", "no", "0" should result in errors instead of being treated as "true".
Also it seems like parsing values for boolean fields into actual boolean values seems like a cleaner way to do it.
Feel free to disregard.

Reply:
We have params that can be `boolean` or `string`, e.g. `backdrop`. Because of that, I'd prefer to disregard this suggestion.